Library/Formula/dex.rb

```diff
@@ -0,0 +1,40 @@
+class Dex < Formula
+  desc "Dextrous text editor"
+  homepage "https://github.com/tihirvon/dex"
+  url "https://github.com/tihirvon/dex/archive/v1.0.tar.gz"
+  head "https://github.com/tihirvon/dex.git"
+
+  sha256 "4468b53debe8da6391186dccb78288a8a77798cb4c0a00fab9a7cdc711cd2123"
+
+  depends_on "homebrew/dupes/ncurses" => :optional
+  depends_on "homebrew/dupes/libiconv" => :optional
+
+  def install
+    args = ["prefix=#{prefix}",
+            "CC=#{ENV.cc}",
+            "HOST_CC=#{ENV.cc}"]
+
+    args << "VERSION=#{version}" if build.head?
+
+    if build.with? "ncurses"
+      inreplace "Makefile", /-lcurses/, "-lncurses"
+
+      ENV.append_to_cflags "-I#{Formula["ncurses"].opt_include}"
```

Comment:
You can remove the ENV changes for `ncurses` and `libiconv` - They're both `keg_only` and when invoked `brew` will shunt them to the front of the `$PATH` and apply the variables automatically.